From ac419720dbb2e7abf7348a58fba0121ee2221219 Mon Sep 17 00:00:00 2001
From: Michael Haubenwallner <michael.haubenwallner@ssi-schaefer.com>
Date: Wed, 22 May 2019 15:07:13 +0200
Subject: [PATCH 15/19] bits datatypes need to be unsigned

otherwise, with gcc-8 we get this error:
uni_fromuni.cc:554:16: error: narrowing conversion of '4026531840' from 'unsigned int' to 'bits32' {aka 'long int'} inside { } [-Wnarrowing]
     0xF0000000 };

Also, the bits32 datatype needs to be int, as long may be 64bit.
---
 orb/uni_types.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/orb/uni_types.h b/orb/uni_types.h
index a2f4fbd..50e70d0 100644
--- a/orb/uni_types.h
+++ b/orb/uni_types.h
@@ -40,15 +40,15 @@
 #include <sys/types.h>
 #endif
 
-typedef char  bits8;
-typedef short bits16;
-typedef long  bits32;
+typedef unsigned char  bits8;
+typedef unsigned short bits16;
+typedef unsigned int   bits32;
 #if defined(_WIN32) && !defined(__MINGW32__)
-typedef __int64 bits64;
+typedef unsigned __int64 bits64;
 #elif defined (_POCKET_PC)
-typedef __int64 bits64;
+typedef unsigned __int64 bits64;
 #else
-typedef long long bits64;
+typedef unsigned long long bits64;
 #endif
 
 typedef char uni_byte;
-- 
2.19.2

